### PR TITLE
[MM-28063] Cloud Telemetry - Admin Console

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Telemetry code for the Admin Console portion of Cloud Billing.

NOTES:
- There are a handful of telemetry that weren't yet completed because the features aren't built yet. Will be making a follow-up ticket to finish those up.
- Some of the telemetry items (ie. clicking on cloud-specific portions of the console's sidebar, clicking on the header) are invalid. I've made notes in the Google Doc explaining why we can't do them.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28063

### CODE REVIEW

File: components/admin_console/billing/billing_summary.tsx [hunk 1]
```diff
@@ -33,6 +34,7 @@ const noBillingHistory = (
             rel='noopener noreferrer'
             href={CloudLinks.BILLING_DOCS}
             className='BillingSummary__noBillingHistory-link'
+            onClick={() => trackEvent('cloud_admin', 'click_how_billiing_works', {screen: 'subscriptions'})}
```
Comment (Adovenmuehle) [inline components/admin_console/billing/billing_summary.tsx:37]: Typo here (`billiing`)

**Review by Adovenmuehle — CHANGES_REQUESTED** [review]
Just a typo otherwise LGTM

File: components/admin_console/billing/billing_subscriptions.tsx [hunk 2]
```diff
@@ -58,19 +59,19 @@ const BillingSubscriptions: React.FC<Props> = () => {
...
             (async function getAllAnalytics() {
                 await dispatch(getStandardAnalytics());
             }());
         }
+
+        pageVisited('cloud_admin', 'pageview_billing_subscription');
     }, []);
 
+    const [showDanger, setShowDanger] = useState(false);
+    const [showWarning, setShowWarning] = useState(false);
```
Comment (marianunez) [inline components/admin_console/billing/billing_subscriptions.tsx:73]: Not related to this PR, but should we just remove this sample banners?

File: components/admin_console/billing/billing_subscriptions.tsx [hunk 3]
```diff
@@ -5,34 +5,35 @@ import React, {useState, useEffect} from 'react';
...
 
-import PlanDetails from './plan_details';
 import BillingSummary from './billing_summary';
+import PlanDetails from './plan_details';
 
 import './billing_subscriptions.scss';
 
+const onUpgradeMattermostCloud = () => {
+    trackEvent('cloud_admin', 'click_upgrade_mattermost_cloud');
+};
```
Comment (marianunez) [inline components/admin_console/billing/billing_subscriptions.tsx:35]: When clicking upgrade shouldn't it also open the upgrade modal?

File: components/admin_console/billing/payment_info_edit.tsx [hunk 4]
```diff
@@ -96,10 +97,22 @@ const PaymentInfoEdit: React.FC = () => {
...
+                                        target='_new'
+                                        rel='noopener noreferrer'
+                                        href={CloudLinks.BILLING_DOCS}
+                                    >
+                                        <FormattedMessage
+                                            id='admin.billing.subscription.planDetails.howBillingWorks'
+                                            defaultMessage='See how billing works'
+                                        />
+                                    </a>
+                                </>
```
Comment (marianunez) [inline components/admin_console/billing/payment_info_edit.tsx:115]: 👍

**Review by marianunez — APPROVED** [review]
LGTM, just a comment on the opening of the upgrade modal

File: components/admin_console/billing/billing_history.tsx [hunk 5]
```diff
@@ -33,6 +34,7 @@ const noBillingHistorySection = (
             rel='noopener noreferrer'
             href={CloudLinks.BILLING_DOCS}
             className='BillingHistory__noHistory-link'
+            onClick={() => trackEvent('cloud_admin', 'click_billing_history', {screen: 'billing'})}
```
Comment (nickmisasi) [inline components/admin_console/billing/billing_history.tsx:37]: I've been adding the event category to a list of constants (see [here](https://github.com/mattermost/mattermost-webapp/pull/6763/files#diff-f44101ff0bb1f1a11dbbb09428a2cf828fe2ae4b29dac499565b74946bf7d1f9R410))
I'm 0/5 on having to do this everywhere, but we should probably choose one way to do it (either yours or mine) and stick with it.

**Review by nickmisasi — COMMENTED** [review]
One comment about overlap here.

File: components/admin_console/billing/billing_subscriptions.tsx (hunk 3 above)
Comment (devinbinnie) [reply components/admin_console/billing/billing_subscriptions.tsx:35]: Yes, looks like that was missed.

File: components/admin_console/billing/billing_history.tsx (hunk 5 above)
Comment (devinbinnie) [reply components/admin_console/billing/billing_history.tsx:37]: We don't do it anywhere else in the system, you can do a quick search on `trackEvent` and it'll show you.
Not saying we shouldn't use constants, I definitely like the idea of doing so, but for consistencies sake we should probably just not do it until we do it everywhere. 2/5

**Review by Adovenmuehle — APPROVED** [review]

**Review by nickmisasi — APPROVED** [review]

File: components/admin_console/billing/billing_history.tsx (hunk 5 above)
Comment (nickmisasi) [reply components/admin_console/billing/billing_history.tsx:37]: Fair. Mine are already in so will have to go back to adjust at some point

**Pull request merged**